core/src/main/java/tech/pegasys/web3signer/core/config/AzureAuthenticationMode.java

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.web3signer.core.config;
+
+public enum AzureAuthenticationMode {
+  CLIENT_SECRET,
+  MANAGED_IDENTITY
```

Comment:
Wondering if MANAGED_IDENTITY should actually be two modes. Seems a bit magical that system or client managed identity is used based on the presence of the client id.